### DISCUSSION TITLE
Feedback and errors while building from source code

### DISCUSSION THREAD

**Discussion by jolly-g** [sensmetry/sysml-2ls#8] [Q&A] [answered]
**OS :** Windows 11

Hello, 

I'm trying to run SysIDE from source code and I have some feedback and issues. 

1. Error while building from source code

I followed the instruction in the Developer instruction > Build from source code. 
During the `pnpm install`, the installation failed during the `prebuild` script of syside-languageserver. 
If I execute the corresponding command (`node -p \"'export const Version = ' + JSON.stringify(require('./package.json').version) + ';'\" > src/version.ts`) in Powershell, the version.ts file is filed with `export const Version = "0.5.2";` as expected. But if I run the `prebuild` script with pnpm (`pnpm run prebuild`), the version.ts file is created but empty. It's the same with `pnpm install`.
After investigation, the `prebuild` script confuses the `syside-languageserver/node.js` file and NodeJS. Instead of using NodeJS, it tries to use the node.js file. I'm wondering if this issue is only on Windows.

For now, I bypassed this issue by manually creating the version.ts file and deleting the `prebuild` script in the syside-languageserver/package.json. Then I ran `pnpm install`, `pnpm run grammar:generate` and `pnpm run build`.

2. Error while openning a .sysml file

I can now run the extension. While pressing F5, an other VSCode opens without problem. 
But when I open a .sysml file, I got this error in the newly opened VSCode : 
![image](https://github.com/sensmetry/sysml-2ls/assets/149687788/b6550ecb-893a-412f-8493-301a18aaca40)

In the debug console, I got this error : 
![image](https://github.com/sensmetry/sysml-2ls/assets/149687788/fd9841ee-c816-41f8-9967-12a9b14c622a)

I don't know how could I patch this, am I missing something ?

**Comment by daumantas-kavolis-sensmetry** [comment, marked as answer]
Thanks for the feedback, this should be fixed now.
